### PR TITLE
Switch to VFS I/O for notebook upload/download

### PR DESCRIPTION
**Pull request by johnkerl**
Ask from @Shelnutt2 on sc10506

### CODE REVIEW

File: tiledb/cloud/notebook.py [hunk 1]
```diff
@@ -71,7 +71,8 @@ def download_notebook_to_file(
         tiledb_uri,
         storage_credential_name,
     )
-    with open(ipynb_file_name, "w") as handle:
+    vfs = tiledb.VFS()
```
Comment (ihnorton) [inline tiledb/cloud/notebook.py:74]: Probably need to pass the ctx into the VFS constructor, looks like some previous open calls may have been missing it.

Comment (johnkerl) [reply tiledb/cloud/notebook.py:74]: thanks @ihnorton !! can you tell me more about what you mean by "previous" -- ?

Comment (ihnorton) [reply tiledb/cloud/notebook.py:74]: Sorry, I missed that these calls are for local download - please disregard.

File: tiledb/cloud/notebook.py [hunk 2]
```diff
@@ -337,7 +339,9 @@ def _write_notebook_to_array(
 
     contents_as_array = numpy.array(bytearray(ipynb_file_contents, CHARACTER_ENCODING))
 
-    with tiledb.open(tiledb_uri, mode="w", ctx=ctx) as arr:
+    vfs = tiledb.VFS(tiledb.cloud.Ctx().config())
+
+    with vfs.open(tiledb_uri, mode="w", ctx=ctx) as arr:
```
Comment (ihnorton) [inline tiledb/cloud/notebook.py:344]: `vfs.open` returns a FileIO object, not an array, so the assignments below will not work -- this needs to be `tiledb.open` to return an array object.

File: tiledb/cloud/notebook.py [hunk 3]
```diff
@@ -67,7 +67,8 @@ def download_notebook_to_file(
     ipynb_file_contents = download_notebook_contents(
         tiledb_uri,
     )
-    with open(ipynb_file_name, "w") as handle:
```
Comment (ihnorton) [inline tiledb/cloud/notebook.py:70]: This is fine as-is, IMO -- sorry about the distraction with the earlier comment. I missed that this is a local FS operation only.

File: tiledb/cloud/notebook.py [hunk 4]
```diff
@@ -135,7 +136,8 @@ def upload_notebook_from_file(
     :return: TileDB array name, such as "tiledb://janedoe/testing-upload".
     """
 
-    with open(ipynb_file_name, "r") as handle:
+    vfs = tiledb.VFS(tiledb.cloud.Ctx().config())
```
Comment (ihnorton) [inline tiledb/cloud/notebook.py:139]: same as above

**Review by ihnorton — CHANGES_REQUESTED** [review]

File: tiledb/cloud/notebook.py (hunk 2 above)
Comment (johnkerl) [reply tiledb/cloud/notebook.py:344]: sorry @ihnorton it looks like i was rushing through this a bit :(

**Review by ihnorton — APPROVED** [review]

**Comment by johnkerl** [comment]
thanks @ihnorton ! :)

**Pull request merged**